src/6_branch.js

```
@@ -1027,6 +1016,205 @@ Branch.prototype['sendSMS'] = wrap(
+ * with a reminder to call *branch.deepview* first.
+ *
+ * ##### Usage
+ * ```js
+ * // If you're using jQuery, then:
+ * $('a.deepview-cta').click(branch.deepviewCta);
+ *
+ * // Or you can:
+ * document.getElementById('my-elem').onClick = branch.deepviewCta;
+ *
+ * // Or in HTML you can:
+ * <a href='...' onclick='branch.deepviewCta(null, true); return false'>
+ *
+ * // If you wish to dedicate a CTA link only to branch deepviewCta, you can:
+ * branch.deepview(data, option, function(err) {
+ *     if (err) {
+ *         throw err;
+ *     }
+ *     ${'a.deepview-cta').click = branch.deepviewCta(null, false);
+ * });
+ *
+ * // You can call this function any time after branch.deepview() is finished by simply:
+ * branch.deepviewCta();
+ * ```
+ * ___
+ *
+ * # Referral system rewarding functionality
+ * In a standard referral system, you have 2 parties: the original user and the invitee. Our system is flexible enough to handle rewards for all users for any actions. Here are a couple example scenarios:
+ * 1. Reward the original user for taking action (eg. inviting, purchasing, etc)
+ * 2. Reward the invitee for installing the app from the original user's referral link
+ * 3. Reward the original user when the invitee takes action (eg. give the original user credit when their the invitee buys something)
+ *
+ * These reward definitions are created on the dashboard, under the 'Reward Rules' section in the 'Referrals' tab on the dashboard.
+ *
+ * Warning: For a referral program, you should not use unique awards for custom events and redeem pre-identify call. This can allow users to cheat the system.
+ *
+ * ## Retrieve referrals list
+ *
+ */
+/*** +TOC_ITEM #deepviewctaevent-silent &.deepviewCta()& ^ALL ***/
+Branch.prototype['deepviewCta'] = wrap(callback_params.NO_CALLBACK, function(done, event, silent) {
+	if (typeof this._deepviewCta === 'undefined') {
+		if (silent) {
```

if you are a lazy developer who doesn't want to set the action handler to deepviewCta inside deepview callback, instead you prefer to do `onclick="branch.deepviewCta()"` inside HTML directly, and you don't want to handle the "call deepview first before you call deepviewCta" error, then you sent `silent` to true and if the user clicks the CTA before deepview() call is finished, it will fail silently.

Or do you think we should always threw error when deepview() call is not finished?